python/plugins/processing/core/ProcessingConfig.py

```diff
@@ -83,6 +84,10 @@ def initialize():
             ProcessingConfig.tr('Use only selected features'), True))
         ProcessingConfig.addSetting(Setting(
             ProcessingConfig.tr('General'),
+            ProcessingConfig.IGNORE_NULL_GEOMS,
+            ProcessingConfig.tr('Ignore features with null geometries'), True))
```

Comment:
@nyalldawson @m-kuhn: a bit rationale for this: Most algorithms required geoms (mainly those that do some computational geometry stuff, like dissolve, convex hull, intersect, etc). Most of those do not explicitly check that, so they break. My idea with this check is done by Processing, so those algorithms and all new ones wont have to add that extra code.

If the alg can accept null geoms, then it is very easy to indicate that in the algorithm, just pass ignoreNullGeoms=False to the feature iterator.

So yes, it is something algorithm-dependent, as @nyalldawson mentions, but this is the easiest way IMHO to handle that in algorithms.

Let me know what you think.